### DISCUSSION TITLE
classes/class_inputeventjoypadmotion

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#192] [User-contributed notes]
# classes/class_inputeventjoypadmotion

Inherits: InputEvent< Resource< RefCounted< Object Represents axis motions (such as joystick or analog triggers) from a gamepad. Description: Stores information about joystick motions. One InputEve...

https://docs.godotengine.org/en/4.3/classes/class_inputeventjoypadmotion.html

**Comment by Faecous** [comment]
If you want to get the joystick position without getting the value from `event.get_axis_value()`, you can use `Input.get_vector`. You just have to define the axis in your Input map first, as `Input.get_vector` takes four string names as arguments.

**Comment by JamesMcVitty** [comment]
It could be worth clarifying that `axis_value` is also the property you would reference when getting or setting the positive or negative direction of an `InputEventJoypadMotion` in the `InputMap`.

For example, if you create a `"forwards"` action  in the Input Map in Project Settings, and add the `event` "Joypad Axis 1 - (Left Stick Up)", you could reference this in code, and it would correspond to `axis = 1` and `axis_value = -1.0`. This could be handy to know for anyone setting up input remapping or other dynamic or accessible input systems.
